cluster/clustered_service_agent.go

```diff
@@ -663,12 +665,28 @@ func (agent *ClusteredServiceAgent) awaitRecordingId(sessionId int32) (int64, er
 			}
 			return false
 		})
+		if _, err := arch.PollForErrorResponse(); err != nil {
+			return NullValue, counterId, err
+		}
 		if counterId != NullValue {
-			return recId, nil
+			return recId, counterId, nil
 		}
 		agent.Idle(0)
 	}
-	return NullValue, fmt.Errorf("timed out waiting for recordingId for sessionId=%d", sessionId)
+}
+
+func (agent *ClusteredServiceAgent) awaitRecordingComplete(recordingId int64, position int64, counterId int32, arch *archive.Archive) error {
+	for agent.counters.GetCounterValue(counterId) < position {
+		agent.Idle(0)
+		if _, err := arch.PollForErrorResponse(); err != nil {
+			return err
+		}
+
+		if archive.IsRecordingActive(agent.counters, counterId, recordingId) {
```

Comment:
Is this supposed to be `archive.IsRecordingActive` or `!archive.IsRecordingActive`?

Reply:
Thanks! Nice catch. I shouldn't commit code at 1am, such silly error

Reply:
hoagotosleep.com